src/App.js

```diff
@@ -14,11 +14,90 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends Component {
+  state = {
+    goods: goodsFromServer,
+    showList: false,
+  }
+
+toggleListHandler = () => {
+  this.setState(state => ({
+    showList: !state.showList,
+  }));
+};
+
+reverseList = () => {
+  this.setState(state => ({
+    goods: [...state.goods].reverse(),
+  }));
+};
+
+sortList = () => {
+  this.setState(state => ({
+    goods: [...state.goods].sort(),
+  }));
+}
+
+resetList = () => {
+  this.setState({
+    goods: goodsFromServer,
+  });
+};
+
+sortByLength = () => {
+  this.setState(state => ({
+    goods: [...state.goods].sort((a, b) => a.length - b.length),
+  }));
+};
+
+render() {
+  return (
+    <div className="app">
+      <h1>Goods</h1>
+      <button
+        className={this.state.showList && 'visibility'}
```

Comment:
```suggestion
      {!showList && <button
```
Use conditional rendering instead of setting `display:block`